context/user_plane_information.go

```diff
@@ -354,13 +355,73 @@ func (upi *UserPlaneInformation) InsertSmfUserPlaneNode(name string, node *facto
 	return nil
 }
 
-// update UPF info
-func (upi *UserPlaneInformation) UpdateSmfUserPlaneNode(name string, node *factory.UPNode) error {
-	//What all to update- nssai/dnn info
-	logger.UPNodeLog.Infof("UPNode[%v] to update, content[%v]\n", name, node)
-	logger.UPNodeLog.Debugf("content of map[UPNodes] %v \n", upi.UPNodes)
-	upi.DeleteSmfUserPlaneNode(name, node)
-	upi.InsertSmfUserPlaneNode(name, node)
+func (upi *UserPlaneInformation) UpdateSmfUserPlaneNode(name string, newNode *factory.UPNode) error {
```

Comment:
As this PR adds a lot of logic to this function, can you also add a docstring?

Reply:
Done